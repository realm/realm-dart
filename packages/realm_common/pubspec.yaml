name: realm_common
description: >-
             Hosts the common code shared between realm, realm_dart and realm_generator packages.
             This package is part of the official Realm Flutter and Realm Dart SDKs.

version: 3.2.0

homepage: https://www.realm.io
repository: https://github.com/realm/realm-dart
issue_tracker: https://github.com/realm/realm-dart/issues

environment:
  sdk: ^3.3.0

dependencies:
  collection: ^1.18.0
  objectid: ^3.0.0
  sane_uuid: ^1.0.0

dev_dependencies:
  lints: ^3.0.0
