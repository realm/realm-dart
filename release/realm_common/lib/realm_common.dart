// Copyright 2021 MongoDB, Inc.
// SPDX-License-Identifier: Apache-2.0

export 'src/realm_common_base.dart';
export 'src/realm_types.dart';

export 'package:objectid/objectid.dart' show ObjectId;
export 'package:sane_uuid/uuid.dart' show Uuid;
