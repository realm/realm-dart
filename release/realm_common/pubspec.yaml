name: realm_common
description: >-
             Hosts the common code shared between realm, realm_dart and realm_generator packages.
             This package is part of the official Realm Flutter and Realm Dart SDKs.

version: 2.0.0-alpha.5

homepage: https://www.realm.io
repository: https://github.com/realm/realm-dart
issue_tracker: https://github.com/realm/realm-dart/issues

environment:
  sdk: ^3.0.0

dependencies:
  collection: ^1.18.0
  objectid: ^3.0.0
  sane_uuid: ^1.0.0-alpha.5

dev_dependencies:
  lints: ^3.0.0
