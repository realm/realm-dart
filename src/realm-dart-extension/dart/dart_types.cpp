////////////////////////////////////////////////////////////////////////////
//
// Copyright 2016 Realm Inc.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
// http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
////////////////////////////////////////////////////////////////////////////
#pragma clang diagnostic ignored "-Wdocumentation"

#include <cmath>
#include <functional>
#include <map>
#include <string>

#include "dart_api.h"
#include "dart_native_api.h"

#include "js_types.hpp"

#include "dart_types.hpp"

bool ___DartError::use_native_exceptions;

Dart_Handle operator ||(const Dart_Handle& dartResult, ___DartError error) {
	if (Dart_IsError(dartResult)) {
		auto err = Dart_GetError(dartResult);
		//Dart_PropagateError(dartResult);
		if (!___DartError::use_native_exceptions) {
			Dart_ThrowException(dartResult);
		}
		else {
			throw std::runtime_error(err);
		}
		
	}
	return dartResult;
}

namespace realm {
	namespace dartvm {
		namespace flutter {
			std::string FilesDir;
		}

		Dart_Handle DartCoreLibrary;
		Dart_Handle RealmLibrary;
		std::string RealmPackageName;
		Dart_Handle DartDateTimeType;
		
		Dart_Handle RealmCreateDateTimeFunc;
		Dart_Handle RealmDynamicObjectType;
		Dart_Handle RealmTypeStaticPropertiesType;
		Dart_Handle RealmHelpersType;
	}

	namespace util {
		void to_lower(char& c) {
			c = std::tolower(static_cast<unsigned char>(c));
		}

		void to_lowecase(std::string& string)
		{
			std::for_each(string.begin(), string.end(), to_lower);
		}
	}
}

namespace Dart {

	HandleScope::HandleScope() {
		Dart_EnterScope();
	}

	HandleScope::~HandleScope() {
		Dart_ExitScope();
	}

	Dart_Handle GetLibrary(const char* libName) {
		Dart_Handle libraryUrl = Dart_NewStringFromCString(libName) || handleError;
		Dart_Handle library = Dart_LookupLibrary(libraryUrl) || handleError;
		return library;
	}
}

