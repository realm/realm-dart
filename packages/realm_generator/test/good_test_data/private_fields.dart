// ignore_for_file: unused_element, prefer_final_fields, unused_field

import 'package:realm_common/realm_common.dart';

part 'private_fields.realm.dart';

@RealmModel()
class _WithPrivateFields {
  late String _plain;
  int _withDefault = 0;
}
