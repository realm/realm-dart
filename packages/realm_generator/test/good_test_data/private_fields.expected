// GENERATED CODE - DO NOT MODIFY BY HAND

part of 'private_fields.dart';

// **************************************************************************
// RealmObjectGenerator
// **************************************************************************

// ignore_for_file: type=lint
class WithPrivateFields extends _WithPrivateFields
    with RealmEntity, RealmObjectBase, RealmObject {
  static var _defaultsSet = false;

  WithPrivateFields(
    String _plain, {
    int withDefault = 0,
  }) {
    if (!_defaultsSet) {
      _defaultsSet = RealmObjectBase.setDefaults<WithPrivateFields>({
        '_withDefault': 0,
      });
    }
    RealmObjectBase.set(this, '_plain', _plain);
    RealmObjectBase.set(this, '_withDefault', withDefault);
  }

  WithPrivateFields._();

  @override
  String get _plain => RealmObjectBase.get<String>(this, '_plain') as String;
  @override
  set _plain(String value) => RealmObjectBase.set(this, '_plain', value);

  @override
  int get _withDefault => RealmObjectBase.get<int>(this, '_withDefault') as int;
  @override
  set _withDefault(int value) =>
      RealmObjectBase.set(this, '_withDefault', value);

  @override
  Stream<RealmObjectChanges<WithPrivateFields>> get changes =>
      RealmObjectBase.getChanges<WithPrivateFields>(this);

  @override
  Stream<RealmObjectChanges<WithPrivateFields>> changesFor(
          [List<String>? keyPaths]) =>
      RealmObjectBase.getChangesFor<WithPrivateFields>(this, keyPaths);

  @override
  WithPrivateFields freeze() =>
      RealmObjectBase.freezeObject<WithPrivateFields>(this);

  EJsonValue toEJson() {
    return <String, dynamic>{
      '_plain': _plain.toEJson(),
      '_withDefault': _withDefault.toEJson(),
    };
  }

  static EJsonValue _toEJson(WithPrivateFields value) => value.toEJson();
  static WithPrivateFields _fromEJson(EJsonValue ejson) {
    if (ejson is! Map<String, dynamic>) return raiseInvalidEJson(ejson);
    return switch (ejson) {
      {
        '_plain': EJsonValue _plain,
      } =>
        WithPrivateFields(
          fromEJson(_plain),
          withDefault: fromEJson(ejson['_withDefault'], defaultValue: 0),
        ),
      _ => raiseInvalidEJson(ejson),
    };
  }

  static final schema = () {
    RealmObjectBase.registerFactory(WithPrivateFields._);
    register(_toEJson, _fromEJson);
    return const SchemaObject(
        ObjectType.realmObject, WithPrivateFields, 'WithPrivateFields', [
      SchemaProperty('_plain', RealmPropertyType.string),
      SchemaProperty('_withDefault', RealmPropertyType.int),
    ]);
  }();

  @override
  SchemaObject get objectSchema => RealmObjectBase.getSchema(this) ?? schema;
}
